### ISSUE TITLE
TypeError: window.sessionStorage is null

### ISSUE THREAD

**Issue by lencioni** [rackt/history#128]
In our error logs, we occasionally see this error:

> TypeError: window.sessionStorage is null

Here's the stack trace that was logged:

```
TypeError: window.sessionStorage is null
  at ReactUpdates(~/react/lib/ReactEventListener.js?2365**:204:0)
  at ReactUpdates(~/react/lib/ReactUpdates.js?ce09**:94:0)
  at batchingStrategy(~/react/lib/ReactDefaultBatchingStrategy.js?ef70**:62:0)
  at transaction(~/react/lib/Transaction.js?6dff**:136:0)
  at method(~/react/lib/ReactEventListener.js?2365**:73:0)
  at handleTopLevelWithoutPath(~/react/lib/ReactEventListener.js?2365**:93:0)
  at ReactEventListener(~/react/lib/ReactEventEmitterMixin.js?8a20**:34:0)
  at runEventQueueInBatch(~/react/lib/ReactEventEmitterMixin.js?8a20**:18:0)
  at EventPluginHub(~/react/lib/EventPluginHub.js?c6b1**:259:0)
  at forEachAccumulated(~/react/lib/forEachAccumulated.js?b898**:23:0)
  at arr(~/react/lib/EventPluginHub.js?c6b1**:54:0)
  at executeDispatchesAndRelease(~/react/lib/EventPluginHub.js?c6b1**:43:0)
  at EventPluginUtils(~/react/lib/EventPluginUtils.js?0958**:102:0)
  at executeDispatch(~/react/lib/EventPluginUtils.js?0958**:79:0)
  at ReactErrorUtils(~/react/lib/ReactErrorUtils.js?6b31**:27:0)
  at func(./app/assets/components/Button/index.jsx:217:12)
  at _this(./app/assets/components/pages/PetitionPageController/index.jsx:128:4)
  at afterAuth(./app/assets/javascripts/lib/after_auth.jsx:215:6)
  at RedirectTo(./app/assets/javascripts/lib/redirect_to.js:22:6)
  at History(~/history/lib/createHistory.js:154:0)
  at transitionTo(~/history/lib/createHistory.js:139:0)
  at confirmTransitionTo(~/history/lib/createHistory.js:113:0)
  at _AsyncUtils(~/history/lib/AsyncUtils.js:25:0)
  at next(~/history/lib/AsyncUtils.js:21:0)
  at done(~/history/lib/AsyncUtils.js:12:0)
  at callback(~/history/lib/createHistory.js:127:0)
  at callback(~/history/lib/createHistory.js:143:0)
  at finishTransition(~/history/lib/createBrowserHistory.js:91:0)
  at _DOMStateStorage(~/history/lib/DOMStateStorage.js:23:0)
```

User agent string: "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:34.0) Gecko/20100101 Firefox/34.0" which translates to Firefox 34 on Windows 7 I believe.

This seems like it was maybe an anomaly especially since we rarely see it, but I figured I'd put it out there in case it is helpful for someone else. Any ideas?

**Comment by terinjokes** [comment]
For someone trying to reproduce this, I've seen this error active when Firefox and Safari are in private browsing modes, and when Safari has cookies disabled.

**Comment by taion** [comment]
Care to add a PR that additionally handles the case where `window.sessionStorage` isn't there, then?

**Comment by terinjokes** [comment]
@taion I can do that tonight/tomorrow, sure.

**Comment by taion** [comment]
Thanks. Looking forward to it.

**Comment by taion** [comment]
@terinjokes Did you get a chance to get to this? I'm going to merge this into #123 if not.

**Comment by terinjokes** [comment]
Nope. Got busy and then totally forgot. Sorry :(

**Comment by mjackson** [comment]
This sounds like a dup of #120. You shouldn't see this particular error any more in your logs since we merged #120 (version > 1.13.1) but we still need to address DOM state storage failures better. That's what #123 is for.
